src/storage.cc

```diff
@@ -46,10 +46,95 @@ void Storage::InitFrom(cyclus::QueryableBackend* b) {
+                          active_buying_min, active_buying_max));
+  }
+  else {
+    throw cyclus::ValueError("Invalid active buying frequency type");}
+
+  /// set up dormant buying distribution
+  if (dormant_buying_frequency_type == "Fixed") {
+    dormant_dist_ = boost::shared_ptr<cyclus::FixedIntDist>(new cyclus::FixedIntDist(dormant_buying_val));
+  }
+  else if (dormant_buying_frequency_type == "Uniform") {
+    if ((dormant_buying_min == -1) || (dormant_buying_max == -1)) {
+      throw cyclus::ValueError("Invalid dormant buying frequency range. Please provide both a min and max value.");
+    }
+    dormant_dist_ = boost::shared_ptr<cyclus::UniformIntDist>(new cyclus::UniformIntDist(dormant_buying_min, dormant_buying_max));
+  }
+  else if (dormant_buying_frequency_type == "Normal") {
+    if ((dormant_buying_mean == -1) || (dormant_buying_stddev == -1)) {
+      throw cyclus::ValueError("Invalid dormant buying frequency range. Please provide both a mean and standard deviation value.");
+    }
+    if (dormant_buying_min == -1) {dormant_buying_min = 1;}
+    if (dormant_buying_max == -1) {
+      dormant_buying_max = std::numeric_limits<int>::max();}
+    dormant_dist_ = boost::shared_ptr<cyclus::NormalIntDist>(new cyclus::NormalIntDist(dormant_buying_mean, dormant_buying_stddev,
+                          dormant_buying_min, dormant_buying_max));
+  }
+  else {
+    throw cyclus::ValueError("Invalid dormant buying frequency type");}
+
+  /// set up buying size distribution
+  if (buying_size_type == "Fixed") {
+    size_dist_ = boost::shared_ptr<cyclus::FixedDoubleDist>(new cyclus::FixedDoubleDist(buying_size_val));
+  }
+  else if (buying_size_type == "Uniform") {
+    if ((buying_size_min == -1) || (buying_size_max == -1)) {
+      throw cyclus::ValueError("Invalid buying size range. Please provide both a min and max value.");
+    }
+    size_dist_ = boost::shared_ptr<cyclus::UniformDoubleDist>(new cyclus::UniformDoubleDist(buying_size_min, buying_size_max));
+  }
+  else if (buying_size_type == "Normal") {
+    if ((buying_size_mean == -1) || (buying_size_stddev == -1)) {
+      throw cyclus::ValueError("Invalid buying size range. Please provide both a mean and standard deviation value.");
+    }
+    if (buying_size_min == -1) {buying_size_min = 0;}
+    if (buying_size_max == -1) {buying_size_max = 1;}
+    size_dist_ = boost::shared_ptr<cyclus::NormalDoubleDist>(new cyclus::NormalDoubleDist(buying_size_mean, buying_size_stddev,
+                             buying_size_min, buying_size_max));
+  }
+  else {
+    throw cyclus::ValueError("Invalid buying size type");}
+}
+
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 void Storage::EnterNotify() {
   cyclus::Facility::EnterNotify();
-  buy_policy.Init(this, &inventory, std::string("inventory"), throughput, active_buying, dormant_buying);
+ 
+  SetUpBuyPolicy();
```

Comment:
pedantic request: can we rename this function to something like `InitBuyPolicyParameters()` since it doesn't actually setup the buy policy - that happens in the next line.

Alternatively, we could have `buy_policy.Init()` called from within this function, but I don't have a good sense of whether that has broader implications.